### PR TITLE
Add/accept invite endpoint

### PR DESCRIPTION
**Pull request by enejb**
Makes the accept invite endpoint work. We do not need the accept api endpoint from endpint to live on the Jetpack side but the following PR is made for convenient testing. Of the invite flow. 

It contains content from the following PRs. 
https://github.com/Automattic/jetpack/pull/7071
https://github.com/Automattic/jetpack/pull/7068
https://github.com/Automattic/jetpack/pull/7060

#### Changes proposed in this Pull Request:
* 

#### Testing instructions:
* Use the following calypso PR https://github.com/Automattic/wp-calypso/pull/13309
* The .com PR and load this branch. 
D5335-code and D5206-code to your .com Sandbox. 

Make sure that your Jetpack side is able to talk to you .com sandbox via the api by setting the following constant in the wp-config of your jetpack site. 

define('JETPACK__WPCOM_JSON_API_HOST', 'somethingsomething.wordpress.com' );

Do the invites work as expected? 
Sandbox the public-api so that the browser is talking to the new api endpoints that are only avilble on your sandbox. 

Send the invite and update the url to point to calypso.localhost:3000 and open it in an incognto mode. 

Does the user get added as expected?

### CODE REVIEW

**Comment by lamosty** [comment]
Code here looks good to me.

File: class.json-api-endpoints.php [hunk 1]
```diff
@@ -267,11 +267,32 @@ function input( $return_default_values = true, $cast_and_filter = true ) {
 			break;
 		}
 
-		if ( !$cast_and_filter ) {
-			return $return;
+		if ( isset( $this->api->query['force'] ) 
+		    && 'secure' === $this->api->query['force']
+		    && isset( $return['secure_key'] ) ) {
+			$this->api->post_body = $this->get_secure_body( $return['secure_key'] );
```
Comment (artpi) [inline class.json-api-endpoints.php]: It would be really cool to have a comment here since later when we forget how this works it will be hard to follow what is going on

**Comment by roccotripaldi** [comment]
One thing I noticed, when I invited a wpcom user with 2FA enabled account, after accepting the invite, the page got hung up.
The invite was accepted successfully, but I never got the confirmation notification, nor was I redirected as expected. This may be because my wpcom sandbox was being very slow. 
But someone else may want to re-test with a 2FA account to double check.
